### PR TITLE
clientv3: document context to "KeepAlive" API

### PR DESCRIPTION
**Pull request by gyuho**
Address https://github.com/coreos/etcd/issues/8993.

### CODE REVIEW

File: clientv3/lease.go [hunk 1]
```diff
@@ -114,6 +114,12 @@ type Lease interface {
 	Leases(ctx context.Context) (*LeaseLeasesResponse, error)
 
 	// KeepAlive keeps the given lease alive forever.
+	// When the context "ctx" is canceled or timed out,
+	// returned "LeaseKeepAliveResponse" channel is closed,
+	// and "LeaseKeepAliveResponse" from this closed channel is nil.
+	// If the context is "context.Background/TODO",
```
Comment (xiang90) [inline clientv3/lease.go]: probably wont say anything about the TODO one. line 116 already explains it. we might want to talk about when an error can be returned though.

**Comment by gyuho** [comment]
@xiang90 Fixed. PTAL.

File: clientv3/lease.go [hunk 2]
```diff
@@ -114,6 +114,9 @@ type Lease interface {
 	Leases(ctx context.Context) (*LeaseLeasesResponse, error)
 
 	// KeepAlive keeps the given lease alive forever.
+	// If the context "ctx" is canceled or timed out,
+	// returned "LeaseKeepAliveResponse" channel is closed,
+	// and "LeaseKeepAliveResponse" from this closed channel is nil.
 	KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAliveResponse, error)
```
Comment (xiang90) [inline clientv3/lease.go]: we should document about the error? for example, what happens if the client failed to send keepalive for a few minutes, and when it reconnected the lease is already expired?

**Comment by gyuho** [comment]
@xiang90 PTAL. Picked up the documentation from https://github.com/coreos/etcd/pull/7866.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**